src/parser.ts

```diff
@@ -296,19 +332,30 @@ export function collapseTerm(tokens: ExprElement[]): ExprElement {
     }
   }
 
+  // Replace all operator minuses, not preceded by numbers, with functions. Each function takes only one argument,
+  // the next token in sequence.
+  // Move backwards to correctly handle nested expressions. For example, " - - a" should be parsed as function "−" with
+  // argument [function "-" with argument ["a"]].
+  for (let i = tokens.length - 1; i >= 0; i--) {
+    // Treat ± as a minus.
+    if (isOperator(tokens[i], '− ±')) {
+      if (tokens[i - 1] instanceof ExprNumber) continue;
+
+      tokens.splice(i, 2, new ExprFunction('−', [tokens[i + 1]]));
+    }
+  }
+
   // Match multiplication operators.
   tokens = findAssociativeFunction(tokens, '× * ·', true);
 
-  // Match - and ± operators, including a unary -/± at the start of an expression.
-  if (isOperator(tokens[0], '− ±')) {
-    tokens.splice(0, 2, new ExprFunction((tokens[0] as ExprOperator).o, [tokens[1]]));
-  }
-  findBinaryFunction(tokens, '− ±');
+  findBinaryFunction(tokens, '- −');
```

Comment:
I don't think it should be a part of this PR but perhaps some day we should rename this function and `findBinaryMinusFunctions` to have more descriptive names, and/or add a doc comment to the former. To help distinguish what they do.